MCAntiMalware-Core/src/main/java/optic_fusion1/antimalware/scanner/Scanner.java

```diff
@@ -128,7 +128,7 @@ private void scanZip(Path zippedFile) {
       try ( PrintWriter pw = new PrintWriter(file)) {
         pw.write(zippedFile.toString());
       }
-      for (File foundDir : destination.listFiles()) {
+      for (File foundDir : Objects.requireNonNull(Objects.requireNonNull(destination.listFiles()))) {
```

Comment:
I don't really see why this is needed? The destination folder is always created and iirc if there's no files in the directory nothing will happen